tests/schemabuilder/builder_test.py

```diff
@@ -16,227 +18,219 @@
 
-    def test_schema_update_with_query(self):
-        schema = build_schema_batch([self.exchange_wo_query], UpdateMode.GEN)
+    operation = schema.paths[expected_path_name].get
+    assert operation.parameters == []
 
-        operation = schema.paths[self.expected_path_name].get
-        assert operation.parameters == []
+    updated_schema = update_openapi(schema, query_exchange, UpdateMode.GEN)
 
-        updated_schema = update_openapi(schema, self.exchange, UpdateMode.GEN)
+    operation = updated_schema.paths[expected_path_name].get
+    assert_that(operation.parameters, has_length(2))
 
-        operation = updated_schema.paths[self.expected_path_name].get
-        assert_that(operation.parameters, has_length(2))
+    first_query_param = operation.parameters[0]
 
-        first_query_param = operation.parameters[0]
+    assert not first_query_param.required
 
-        assert not first_query_param.required
 
+text_request = RequestBuilder.from_url("https://example.com/v1")
+text_response = Response(statusCode=200, body="Hello World", headers={}, bodyAsJson=None, timestamp=None)
 
-class TestSchemaTextBody:
-    request = RequestBuilder.from_url("https://example.com/v1")
-    response = Response(statusCode=200, body="Hello World", headers={}, bodyAsJson=None, timestamp=None)
-    exchange: HttpExchange = HttpExchange(request=request, response=response)
+@pytest.fixture
+def text_exchange():
+    return HttpExchange(request=text_request, response=text_response)
 
-    def test_build_string_body(self):
-        schema = build_schema_batch([self.exchange], UpdateMode.GEN)
-        response_content = schema.paths['/v1'].get.responses['200'].content
+def test_build_string_body(text_exchange):
+    schema = build_schema_batch([text_exchange], UpdateMode.GEN)
+    response_content = schema.paths['/v1'].get.responses['200'].content
 
-        assert_that(response_content, has_key("text/plain"))
+    assert_that(response_content, has_key("text/plain"))
 
-        media_type = response_content["text/plain"]
+    media_type = response_content["text/plain"]
 
-        assert media_type.schema._type == "string"
+    assert media_type.schema._type == "string"
 
-class TestShemaInReplayMode:
+def test_schema_in_replay_mode():
+    reqs = []
+    with open('tests/server/mock/callbacks/opbank/recordings/recording.jsonl','r') as rr: 
+        reqs = rr.read().split('\n')
 
-    def test_schema_in_replay_mode(self):
-        reqs = []
-        with open('tests/server/mock/callbacks/opbank/recordings/recording.jsonl','r') as rr: 
-            reqs = rr.read().split('\n')
+    reqs = [HttpExchangeBuilder.from_dict(json.loads(r)) for r in reqs if r != '']
+    r = build_schema_batch(reqs, UpdateMode.REPLAY)
+    # this schema has four recordings, of which two correspond to /v1/payments/confirm
+    assert 2 == len(r.paths['/v1/payments/confirm'].post.responses['201'].content['application/json'].schema.oneOf)
```

Comment:
Such complex asserts without any assertion error message can be hard for others to debug so that's why I've used hamcrest matchers when possible. But this is fine of course.

Reply:
Good call, I need to convert these to hamcrest. The issue is that VS code for me underlines all the matchers in red, so i need to figure out why...